`pulumi preview --import-file...` against a stack with resources produces a panic

### What happened?

A panic occurs when running `pulumi preview --import-file ...` against a stack that contains resources.

### Example

`pulumi up` the below program to ensure the stack has resources:
```
import * as awsx from "@pulumi/awsx";

new awsx.ec2.Vpc("vpc", {
    cidrBlock: "10.100.0.0/16"
});
```

next, execute `pulumi preview --import-file some-json.json`

panic:
```
panic: runtime error: index out of range [0] with length 0

goroutine 65 [running]:
main.newPreviewCmd.func1.buildImportFile.func1()
	/home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/preview.go:91 +0x140a
github.com/pulumi/pulumi/sdk/v3/go/common/promise.Run[...].func1()
	/home/runner/work/pulumi/pulumi/sdk/go/common/promise/promise.go:150 +0x2c
created by github.com/pulumi/pulumi/sdk/v3/go/common/promise.Run[...] in goroutine 1
	/home/runner/work/pulumi/pulumi/sdk/go/common/promise/promise.go:149 +0x90
```

### Output of `pulumi about`

pulumi/pulumi 3.99

### Additional context

cc @Frassle 

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/changelog/pending/20240110--cli-import--import-file-now-omits-empty-resource-lists.yaml
+++ b/changelog/pending/20240110--cli-import--import-file-now-omits-empty-resource-lists.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: cli/import
+  description: --import-file now omits empty resource lists.

--- a/pkg/cmd/pulumi/import.go
+++ b/pkg/cmd/pulumi/import.go
@@ -161,7 +161,7 @@ type importSpec struct {
 
 type importFile struct {
 	NameTable map[string]resource.URN `json:"nameTable,omitempty"`
-	Resources []importSpec            `json:"resources"`
+	Resources []importSpec            `json:"resources,omitempty"`
 }
 
 func readImportFile(p string) (importFile, error) {

--- a/pkg/cmd/pulumi/import_test.go
+++ b/pkg/cmd/pulumi/import_test.go
@@ -404,33 +404,36 @@ func TestParseImportFileAutoURN(t *testing.T) {
 func TestImportFileMarshal(t *testing.T) {
 	t.Parallel()
 
-	importFile := importFile{
-		NameTable: map[string]resource.URN{
-			"foo": "urn:pulumi:stack::proj::foo:bar:a::arb",
-		},
-		Resources: []importSpec{
-			{
-				Name: "bar",
-				Type: "foo:bar:b",
-				ID:   "123",
-			},
-			{
-				Name:      "comp",
-				Type:      "some/comp",
-				Component: true,
-			},
-			{
-				Name:              "thirdParty",
-				Type:              "some:third:party",
-				ID:                "abc123",
-				Parent:            "bar",
-				Version:           "1.2.3",
-				PluginDownloadURL: "https://example.com",
-			},
-		},
-	}
+	t.Run("initial", func(t *testing.T) {
+		t.Parallel()
 
-	expected := `{
+		importFile := importFile{
+			NameTable: map[string]resource.URN{
+				"foo": "urn:pulumi:stack::proj::foo:bar:a::arb",
+			},
+			Resources: []importSpec{
+				{
+					Name: "bar",
+					Type: "foo:bar:b",
+					ID:   "123",
+				},
+				{
+					Name:      "comp",
+					Type:      "some/comp",
+					Component: true,
+				},
+				{
+					Name:              "thirdParty",
+					Type:              "some:third:party",
+					ID:                "abc123",
+					Parent:            "bar",
+					Version:           "1.2.3",
+					PluginDownloadURL: "https://example.com",
+				},
+			},
+		}
+
+		expected := `{
   "nameTable": {
     "foo": "urn:pulumi:stack::proj::foo:bar:a::arb"
   },
@@ -456,10 +459,23 @@ func TestImportFileMarshal(t *testing.T) {
   ]
 }
 `
-	var buffer bytes.Buffer
-	enc := json.NewEncoder(&buffer)
-	enc.SetIndent("", "  ")
-	err := enc.Encode(importFile)
-	require.NoError(t, err)
-	assert.Equal(t, expected, buffer.String())
+		var buffer bytes.Buffer
+		enc := json.NewEncoder(&buffer)
+		enc.SetIndent("", "  ")
+		err := enc.Encode(importFile)
+		require.NoError(t, err)
+		assert.Equal(t, expected, buffer.String())
+	})
+
+	t.Run("omit empty resources list", func(t *testing.T) {
+		t.Parallel()
+
+		importFile := importFile{}
+
+		var buffer bytes.Buffer
+		enc := json.NewEncoder(&buffer)
+		err := enc.Encode(importFile)
+		require.NoError(t, err)
+		assert.NotContains(t, buffer.String(), "resources")
+	})
 }

